source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java

```diff
@@ -114,8 +128,285 @@ static void setup() {
         snake = NPCFactory.createSnake(player, enemies);
         magpie = NPCFactory.createMagpie(player, enemies);
         fish = NPCFactory.createFish(player, enemies);
+
+        friendlyChicken = NPCFactory.createChicken(player, enemies);
+        friendlyFrog = NPCFactory.createFrog(player, enemies);
+        friendlyMonkey = NPCFactory.createMonkey(player, enemies);
+    }
+
+    @BeforeEach
+    void setUpServices() {
+        dialogueBoxService = mock(DialogueBoxService.class);
+        resourceService = mock(ResourceService.class);
+
+        ServiceLocator.registerDialogueBoxService(dialogueBoxService);
+        ServiceLocator.registerResourceService(resourceService);
+    }
+
+
+    @Test
+    void testInitiateDialogue() {
+        String[] testSoundPath = new String[]{"sounds/tiger-roar.wav"};
+        String[][] testHintText = new String[][]{
+                {"Welcome to Animal Kingdom!", "I am Lenny the Lion.", "/cWhich tip do you wanna hear about?/s01What do potions do???/s02How to beat the final boss/s03Nothing. Bye"},
+                {"Potions heals you by (n) HP!", "I hope this helped."},
+                {"Final boss?? That Kangaroo??", "idk"},
+                {"Good luck!"}
+        };
+        NPCFactory.initiateDialogue(testSoundPath, testHintText);
+        verify(dialogueBoxService).updateText(testHintText, -2);
+
+//        for (String soundPath : testSoundPath) {
```

Comment:
Why is this test commented out? Please either make it work with the current game or remove it.